data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -132,8 +132,12 @@ public CloseableIterable<T> filter(CloseableIterable<T> records) {
   }
 
   private List<Predicate<T>> applyEqDeletes() {
+    if (eqDeletePredicates != null) {
+      return eqDeletePredicates;
+    }
     List<Predicate<T>> isInDeleteSets = Lists.newArrayList();
     if (eqDeletes.isEmpty()) {
+      this.eqDeletePredicates = isInDeleteSets;
```

Comment:
I think a better solution is to make `isInDeleteSets` a private field in place of `eqDeletePredicates`. You should just rename the new `eqDeletePredicates` field to `isInDeleteSets` and set it on the line above instead of creating a local variable.

Reply:
Bacause if encounter some exception when loading delete file, it is difficult to judge weather loaded delete file complete when next retry